library/std/src/net/ip.rs

```diff
@@ -1104,9 +1071,9 @@ impl Ipv6Addr {
     #[rustc_const_unstable(feature = "const_ipv6", issue = "76205")]
     #[stable(feature = "rust1", since = "1.0.0")]
     pub const fn segments(&self) -> [u16; 8] {
-        // All elements in `s6_addr` must be big endian.
+        // All elements in `self.octets` must be big endian.
         // SAFETY: `[u8; 16]` is always safe to transmute to `[u16; 8]`.
-        let [a, b, c, d, e, f, g, h] = unsafe { transmute::<_, [u16; 8]>(self.inner.s6_addr) };
+        let [a, b, c, d, e, f, g, h] = unsafe { transmute::<_, [u16; 8]>(self.octets) };
```

Comment:
This might not be safe, since currently the alignment of `self.octets` is 1, but for `[u16; 8]` it is 2, so the octets might be placed on an invalid address? If this is indeed not safe, then the code was previously relying on the implicit fact that `c::in6_addr.s6_addr` had an alignment at least as large as `u16` :open_mouth:

Reply:
It's not transmuting a reference or pointer. It's transmuting into the array itself. So that should be fine regardless of alignment.

Reply:
True! Thanks for helping me double check this.